source/server/listener_impl.cc

```diff
@@ -545,6 +556,12 @@ void ListenerImpl::buildFilterChains() {
 void ListenerImpl::buildSocketOptions() {
   // TCP specific setup.
   if (connection_balancer_ == nullptr) {
+#ifdef WIN32
+    // On Windows we use the exact connection balancer to dispatch connections
+    // from worker 1 to all workers. This is a perf hit but it is the only way
+    // to make all the workers do work.
+    connection_balancer_ = std::make_shared<Network::ExactConnectionBalancerImpl>();
```

Comment:
There is a cost to using the ExactConnectionBalancer: a mutex is needed for accepting each connection.

If this is the fix, I think it would be better to add an approximate connection balancer, that just randomly picks a worker to send the connection to `rand % num_workers`.

Reply:
Don't we always need in the random case as well? Both for querying `handlers_.size()` and for getting the `handlers_[rand % num_workers]`?

Reply:
With a small bit of re-work, this should be doable without a lock. `handlers_.size()` should equal `num_workers`, and the code wouldn't need to call any methods on the elements in `handlers_`, just return a reference.

This may be worthy of a followup PR instead of doing it in this one. It's also a nice-to-have. ExactBalancer will work, it just may run into a perf bottleneck for high connection rates.

Reply:
That was a bit confusing because I also thought `handlers_.size()` should equal `num_workers` and number of workers should remain constant for the lifetime of the application, right?

Reply:
yeah I would recommend doing the approximate balancer as follow up, but it might be nice to add a follow up. I would also add an INFO or WARN log that this is being force enabled on Windows.

Reply:
Added a warning and a TODO in 5034e6e7bec0bb960b0a89e0ada66b94ebf62205